kubetest/kops.go

```diff
@@ -414,12 +414,20 @@ func (k kops) Up() error {
 	if k.adminAccess == "" {
 		var b bytes.Buffer
 		err := httpReadWithHeaders(externalIPURL, map[string]string{"Metadata-Flavor": "Google"}, &b)
-		if err != nil || net.ParseIP(strings.TrimSpace(b.String())) == nil {
+
+		count := 0
+		for count < 5 && err != nil && net.ParseIP(strings.TrimSpace(b.String())) == nil {
+			time.Sleep(2 * time.Second)
+			count++
 			b.Reset()
-			if err := httpRead("https://v4.ifconfig.co", &b); err != nil {
-				return err
-			}
+			err = httpRead("https://v4.ifconfig.co", &b)
+			log.Printf("external IP cannot be retrieved: %v\n", err)
 		}
+
+		if err != nil || net.ParseIP(strings.TrimSpace(b.String())) == nil {
+			return fmt.Errorf("external IP cannot be retrieved: %v", err)
```

Comment:
likewise, we could be failing here even if `err` is nil so maybe we can elaborate on the Errorf to include those cases as well. perhaps including `b.String()` in the error message.